### PR TITLE
testing functionality

### PR DESCRIPTION
**Pull request by CasoMateo**
Checking final authentication details

### CODE REVIEW

File: backend/server.py [hunk 1]
```diff
@@ -11,12 +11,15 @@
...
+from bson.binary import UUID
+
+
 
 # checar response model 
 app = FastAPI()
 
 origins = [
-  "http://localhost:3000",
+  "http://localhost:3000"
```
Comment (EduardoSantos7) [inline backend/server.py:22]: This won't let you deploy your code in any cloud or between networks, be aware and think about which type of deployment you will do.

File: backend/server.py [hunk 2]
```diff
@@ -27,7 +30,7 @@
   allow_headers=["*"],
 )
 
-cluster = MongoClient("mongodb+srv://InventoryManager:CasMat2*<>@proyectogabrielpastor.hfvj9.mongodb.net/InventoryManagement?retryWrites=true&w=majority")
+cluster = MongoClient()
```
Comment (EduardoSantos7) [inline backend/server.py]: Now you are don't have aconnection to the DB, please use environment variables '.env' file, if you have doubts about it let me know in class

File: backend/server.py [hunk 3]
```diff
@@ -67,23 +70,36 @@ def expiredDate(_date):
       
   return datetime.date(int(measures[0]), int(measures[1]), int(measures[2])) < datetime.date.today()
 
+def getCookie(cname, ccookies):
```
Comment (EduardoSantos7) [inline backend/server.py:73]: Add test for all this functions

File: backend/server.py [hunk 4]
```diff
@@ -94,7 +110,7 @@ def authorizedUserChief(username, user_chief):
     return False
 
   user = users.find_one({ 'username': username })
-  if user_chief != user.level: 
+  if not user['level']:
```
Comment (EduardoSantos7) [inline backend/server.py]: use get to avoid key error or pre-check for the key

File: backend/server.py [hunk 5]
```diff
@@ -103,65 +119,83 @@ def clearSessionFromDatabase(username):
...
+
+  if not authenticatedUser(getCookie('session_id', request.headers['cookies']), getCookie('username', request.headers['cookies'])) or not authorizedUserChief(getCookie('username', request.headers['cookies']), getCookie('user_chief', request.headers['cookies'])):
+    raise HTTPException(status_code=401, detail="Unauthorized")
+   # check if given header cookies match database cookies 
+  
+  usernames = []
+  all_users = [user for user in users.find()]
+  
+  for user in all_users: 
+    usernames.append(user['username'])
```
Comment (EduardoSantos7) [inline backend/server.py]: same here try to avoid key errors

File: backend/server.py [hunk 6]
```diff
@@ -238,42 +281,64 @@ def addMedicine(medicine: Medicine, session_id: Optional[str] = Cookie(None),
...
+    raise HTTPException(status_code=401, detail="Unauthorized") 
   
-  
-   
   content = {'addedTo': False}
+  medicine = medicines.find_one({'_id': ObjectId(attributes.medicine_id)})
+  if attributes.quantity > 0 and attributes.expiry:
+
+    for badge in range(len(medicine['badges'])): 
+      if medicine['badges'][badge]['date'] == attributes.expiry:
```
Comment (EduardoSantos7) [inline backend/server.py:295]: don't use multiple direct accesses

File: backend/server.py [hunk 7]
```diff
@@ -238,42 +281,64 @@ def addMedicine(medicine: Medicine, session_id: Optional[str] = Cookie(None),
...
+  medicine = dict(medicines.find_one({'_id': ObjectId(attributes.medicine_id)})) 
+  
+  if attributes.quantity > 0 and attributes.expiry:
 
-  if attributes.quantity > 0 and attributes.expiry and not expiredDate(attributes.expiry):
-    medicines.update_one({ '$and': [{ '_id': attributes.medicine_id }, 
-    { 'badges': { '$elemMatch': {'$and': [{ 'quantity': { 'ge': attributes.quantity } }, 
-    {'date': attributes.expiry}] } }}]}, { '$inc': { 'badges.$.quantity': -attributes.quantity } })
+    for badge in range(len(medicine['badges'])): 
+      if medicine['badges'][badge]['date'] == attributes.expiry and medicine['badges'][badge]['quantity'] >= attributes.quantity:
```
Comment (EduardoSantos7) [inline backend/server.py:329]: same here

File: frontend/src/components/Home.jsx [hunk 8]
```diff
@@ -1,32 +1,57 @@
...
  
   // replantear lógica
-
-  // const { token, renderModifyUsers, logout, username } = useContext(TokenContext);
+  function getCookie(cname) {
+    let name = cname + "=";
+    let decodedCookie = decodeURIComponent(document.cookie);
+    let ca = decodedCookie.split(';');
+    for(let i = 0; i <ca.length; i++) {
+      let c = ca[i];
```
Comment (EduardoSantos7) [inline frontend/src/components/Home.jsx]: please use explicit names (c is not one)

**Review by EduardoSantos7 — CHANGES_REQUESTED** [review]
Do not upload .pyc  files

File: backend/server.py (hunk 3 above)
Comment (CasoMateo) [reply backend/server.py:73]: Added a separate tests file with all of these

**Pull request merged**